pkg/ui/workspaces/cluster-ui/src/statementDetails/planDetails/planDetails.tsx

```diff
@@ -89,6 +117,82 @@ function renderExplainPlan(
         All Plans
       </Button>
       <SqlBox value={explainPlan} size={SqlBoxSize.large} />
+      {hasInsights &&
+        renderInsights(
+          plan.stats.index_recommendations,
+          plan,
+          statementFingerprintID,
+          sortSetting,
+          onChangeSortSetting,
+        )}
     </div>
   );
 }
+
+function formatIdxRecommendations(
+  idxRecs: string[],
+  plan: PlanHashStats,
+  statementFingerprintID: string,
+): InsightRecommendation[] {
+  const recs = [];
+  for (let i = 0; i < idxRecs.length; i++) {
+    const rec = idxRecs[i];
+    let idxType: InsightType;
+    const t = rec.split(" : ")[0];
+    switch (t) {
+      case "creation":
+        idxType = "CREATE_INDEX";
+        break;
+      case "replacement":
+        idxType = "REPLACE_INDEX";
+        break;
+      case "drop":
+        idxType = "DROP_INDEX";
+        break;
+    }
+    const idxRec: InsightRecommendation = {
+      type: idxType,
+      database: plan.metadata.databases[0],
+      table: "",
+      index_id: 0,
+      query: rec.split(" : ")[1],
```

Comment:
Rather than splitting EXPLAIN-oriented recommendation string, I'd prefer we store just the SQL string in the `statement_statistics` table. This will make this much less hacky and error prone.

`IndexRecommendationSet.Output` could return a `[]string` consisting of just the SQL statements, like `[]string{"CREATE INDEX ON t (a)"}`, and store than in instrumentation. We can then format those strings however we'd like for `EXPLAIN`, and store them as-is in `statement_statistics` to make the UI formatting simpler.

What do you think?

Reply:
The parsing from the instrumentation to the statistics is actually done in another location here:
https://github.com/cockroachdb/cockroach/blob/4655f9765f43f95d910497c3afbed28eef8ca786/pkg/sql/idxrecommendations/idx_recommendations.go#L49-L69
I do have tests in place in case the format changes, but it would be great if the instrumentation could have just the queries. 
The reason why I did this way is because I want to show on the UI if it's creation or replacement. The other way could be saving just the string and then when displaying on the ui checking if there is a single create is `creation` and if it has create and drop it would be a `replacement`. Is there any other option (besides creation and replacement) you can think here?